doc/user_guide/libraries.rst

```diff
@@ -0,0 +1,192 @@
+<https://github.com/stfc/PSyclone/tree/master/lib/nan_test>`__.
+For detailed instructions on how to build and use these libraries
+please refer to their specific ``README.md`` documentation.
+
+.. _libraries-dependencies:
+
+Dependencies
+------------
+
+Building and using the wrapper libraries requires that PSyclone be installed
+on the host system, see section :ref:`Getting Going <getting-going>`. A
+Fortran compiler (e.g. Gnu Fortran compiler, ``gfortran``, is free and easily
+installed) and Gnu Make are also required.
+
+The majority of wrapper libraries use `Jinja
+<https://pypi.org/project/Jinja/>`_ templates to create PSyData-derived
+classes (please refer to :ref:`dev_guide:psy_data` and :ref:`dev_guide:jinja`
+for full details about the PSyData API).
+
+Compilation of ``extract``, ``nan_test``, ``read_only`` and some of the
+profiling wrapper libraries depends on infrastructure libraries relevant
+to the API they are used for. :ref:`LFRic API <dynamo0.3-api>` uses the
+LFRic infrastructure and :ref:`GOcean1.0 <gocean1.0-api>` uses the
+``dl_esm_inf`` `library <https://github.com/stfc/dl_esm_inf>`_.
+
+Some libraries require NetCDF for compilation. Installation of NetCDF is
+described in details in the `hands-on practicals documentation
+<https://github.com/stfc/PSyclone/tree/master/tutorial/practicals#netcdf-library-lfric-examples>`_.
+
+Profiling wrapper libraries that depend on external tools (e.g. dl_timer)
+require these tools be installed and configured beforehand.
+
+.. _libraries-compilation:
+
+Compilation
+-----------
+
+Each library is compiled with ``make`` using the provided ``Makefile`` that
+has configurable options for compiler flags and locations of dependencies.
+
+As in case of :ref:`examples <examples-compilation>`, ``F90`` and
+``F90FLAGS`` specify the compiler and compilation flags to use. The default
+value for ``F90`` is ``gfortran``.
+
+Locations of the top-level ``lib`` directory and the required Jinja templates
+are specified with the ``PSYDATA_LIB_DIR`` and ``LIB_TMPLT_DIR`` variables.
+For testing purposes their default values are set to relative paths to the
+respective directories in the PSyclone repository.
+
+The location of the infrastructure libraries for LFRic and GOcean1.0
+applications can be configured with the variable ``INF_DIR``. Their default
+values are set to relative paths to the locations of these libraries in the
+PSyclone repository. However, the infrastructure libraries are not available
+in a PSyclone installation so in this case ``INF_DIR`` **must be set** to
+the exact path to where the infrastructure can be found.
+
+Profiling wrapper libraries that depend on external tools have specific
+variables that configure paths to where these libraries are located in a
+user environment.
+
+For more information on how to build and configure a specific library
+please refer to its ``README.md`` documentation.
+
+Similar to compilation of the :ref:`examples <examples-compilation>`, the
```

Comment:
I just realise that we now have an inconsistency ... the examples use INF_DIR for gocean, but still LFRIC_DIR for lfric :( More discussion about this further down.

Reply:
We agreed via Teams chat to use ``GOCEAN_INF_DIR`` for GOcean and ``LFRIC_INF_DIR`` for LFRic infrastructure paths, respectively, instead of ``INF_DIR`` for all. I changed this in all relevant `lib` Makefiles and README documentation. I also updated the UG chapters `psy_data.rst` and `libraries.rst`. We agreed to correct examples and tutorial in #1209.

Reply:
Thanks!